client/my-sites/checkout/composite-checkout/components/wp-checkout-order-review.tsx

```diff
@@ -90,17 +94,25 @@ export default function WPCheckoutOrderReview( {
 	const { responseCart, removeCoupon, couponStatus } = useShoppingCart();
 	const isPurchaseFree = responseCart.total_cost_integer === 0;
 
-	const firstDomainItem = responseCart.products.find(
-		( product ) => isDomainTransfer( product ) || isDomainRegistration( product )
-	);
-	const domainUrl = firstDomainItem ? firstDomainItem.meta : siteUrl;
+	const selectedSiteData = useSelector( getSelectedSite );
+
+	const primaryDomain = selectedSiteData?.options?.is_mapped_domain
+		? selectedSiteData?.domain
+		: null;
+	const domainUrl =
```

Comment:
Really minor but this is a lot of code in one line. I wonder if it would help future readers if we split it up a bit. There's different ways that could be done, but here's one suggestion:

```js
const firstDomainProduct = responseCart.products.find( ( product ) => isDomainTransfer( product ) || isDomainRegistration( product ) || isDomainMapping( product ) );
const domainUrl = primaryDomain ?? firstDomainProduct?.meta ?? siteUrl;
```

(Aside: there's a `isDomainProduct` function near these others, but it doesn't include domain transfers, so what we have here is probably more clear. 👍 )

Reply:
> here's one suggestion

I definitely like the readability of that more, but had decided to include the `firstDomainProduct` logic in the chain because I figured it would short-circuit and not run the `find()` if `primaryDomain` exists. It's probably not much of a performance gain, so I'm fine with switching it back.

Reply:
That's a fair point. I personally think the readability in this case is worth the performance hit (the cart typically has very few products anyway) but I'll leave the decision to you.

Reply:
Worth mentioning: another possibility (although it's probably overkill here) is to create a separate function to do all this logic and return the domain URL; that way you could have all the efficiency you like and keep the code elegant.